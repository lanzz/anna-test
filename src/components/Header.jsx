import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addChat } from '../actions';

function Header({ addChat }) {
    return (
        <div className="header">
            <button className="add-chat-button" onClick={addChat}>
                Add chat
            </button>
            <h1>An awesome chat app</h1>
        </div>
    );
}

Header.propTypes = {
    addChat: PropTypes.func.isRequired
};

const mapDispatchToProps = { addChat };

export default connect(undefined, mapDispatchToProps)(Header);
