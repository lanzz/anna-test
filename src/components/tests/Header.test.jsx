import React from 'react';
import { render } from 'enzyme';
import Header from '../Header';

it('renders correctly', () => {
    const wrapper = render(
        <Header />
    );
    expect(wrapper).toMatchSnapshot();
});
