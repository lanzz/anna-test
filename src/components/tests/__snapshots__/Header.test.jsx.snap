// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="header"
>
  <button
    class="add-chat-button"
  >
    Add chat
  </button>
  <h1>
    An awesome chat app
  </h1>
</div>
`;
